packages/electron/lib/install.js

```diff
@@ -66,8 +68,30 @@ module.exports = {
     })
   },
 
-  checkExecExistence () {
-    return fs.stat(paths.getPathToExec())
+  checkExecExistence (pathToExec) {
+    return fs.stat(pathToExec)
+  },
+
+  async checkBinaryArchCpuArch (pathToExec, platform, arch) {
+    if (platform === 'darwin' && arch === 'x64') {
+      return Promise.all([
+        // get the current arch of the binary
+        execa('lipo', ['-archs', pathToExec])
```

Comment:
@flotwig using `lipo` here instead of `file` to more easily parse the output. I considered that `lipo` can return a multiple space separated values such as `x64_86 arm64` but considering we don't build a universal binary, and instead only build a single arch, I didn't think it was necessary

Reply:
I played with `lipo` a bit just now, seems like a good solution. Didn't know about this util.

Reply:
I wonder when `-archs` was introduced. `lipo` has been around for a while from what I can see, unsure on `-archs`.